src/pkg/services/m365/api/mail.go

```diff
@@ -540,3 +541,19 @@ func UnwrapEmailAddress(contact models.Recipientable) string {
 
 	return ptr.Val(contact.GetEmailAddress().GetAddress())
 }
+
+// MailCollisionKey constructs a key from the messageable's creation time, subject, and sender.
+// collision keys are used to identify duplicate item conflicts for handling advanced restoration config.
+func MailCollisionKey(item models.Messageable) string {
+	if item == nil {
+		return ""
+	}
+
+	sender := "unknown"
+
+	if item.GetSender() != nil && item.GetSender().GetEmailAddress() != nil {
+		sender = ptr.Val(item.GetSender().GetEmailAddress().GetAddress())
+	}
+
+	return dttm.Format(ptr.Val(item.GetCreatedDateTime())) + ptr.Val(item.GetSubject()) + sender
```

Comment:
is this sufficiently unique? What if mail forwarding is setup and dumps multiple emails into the same inbox (i.e. old account from college forwards to this one)? Do we think createdDateTime will be unique enough? Or are things somehow batched on the server so the created date times kind of end up the same if one gets a lot of messages at once?

Reply:
I'm hedging bets that the created time isn't duplicated for any set of messages in a container.  There's not a lot of sufficiently unique properties we can rely on here otherwise that aren't already an ID.

In fact, thinking about it more, the created time might be _too_ unique.  It only identifies collisions where the original item is still in the folder.  That escapes the case where a restore is performed twice to the same folder.  Ideally, we'd recognize the prior restored value as a duplicate of the backup entry.  But the restored mail is going to have a bumped created date, which means no collisions.

I might switch this up to a concatenation of the subject and all recipients (including cc and bcc), and then probably a hash of that to avoid enormous keys.

Reply:
Okay, I'm now three miles down a weird rabbithole.  What I'm going to do for this PR, since at this time we only need the overall structure and not the specific collision identification cases, is reduce the data used in identification to the ID, and come back to refine it in a later PR.  In the meantime, I've opened an async discussion around the design to ensure we're selecting the best data for collision comparison in respect to corner cases.

Reply:
sounds good, thanks for the update